[2024-06-16T00:23:43.470+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-16T00:23:43.577+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-16T00:23:10+00:00 [queued]>
[2024-06-16T00:23:43.614+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-16T00:23:10+00:00 [queued]>
[2024-06-16T00:23:43.615+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 6
[2024-06-16T00:23:43.665+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): avg_emission_by_year> on 2024-06-16 00:23:10+00:00
[2024-06-16T00:23:43.679+0000] {standard_task_runner.py:63} INFO - Started process 4148 to run task
[2024-06-16T00:23:43.693+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'avg_emission_by_year', 'manual__2024-06-16T00:23:10+00:00', '--job-id', '143', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmpn7uh44lc']
[2024-06-16T00:23:43.700+0000] {standard_task_runner.py:91} INFO - Job 143: Subtask avg_emission_by_year
[2024-06-16T00:23:43.866+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-16T00:23:10+00:00 [running]> on host 6e26965f02f1
[2024-06-16T00:23:44.912+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='avg_emission_by_year' AIRFLOW_CTX_EXECUTION_DATE='2024-06-16T00:23:10+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-16T00:23:10+00:00'
[2024-06-16T00:23:44.914+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-16T00:23:44.916+0000] {summary_analysis.py:33} INFO - starting average pollutants emission calculation
[2024-06-16T00:23:45.026+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/statistical_analysis/summary_analysis.py:36 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2024-06-16T00:23:46.887+0000] {summary_analysis.py:45} ERROR - error calculating average emission
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 39, in avg_emission_by_year
    avg_emision = sub_data.groupby('Year').mean().reset_index()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 9183, in groupby
    return DataFrameGroupBy(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/groupby.py", line 1329, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/grouper.py", line 1043, in get_grouper
    raise KeyError(gpr)
KeyError: 'Year'
[2024-06-16T00:23:46.893+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-16T00:23:46.895+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 39, in avg_emission_by_year
    avg_emision = sub_data.groupby('Year').mean().reset_index()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 9183, in groupby
    return DataFrameGroupBy(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/groupby.py", line 1329, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/grouper.py", line 1043, in get_grouper
    raise KeyError(gpr)
KeyError: 'Year'
[2024-06-16T00:23:46.913+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Air_pollution_data_pipeline, task_id=avg_emission_by_year, run_id=manual__2024-06-16T00:23:10+00:00, execution_date=20240616T002310, start_date=20240616T002343, end_date=20240616T002346
[2024-06-16T00:23:46.938+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 143 for task avg_emission_by_year ('Year'; 4148)
[2024-06-16T00:23:46.970+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-16T00:23:47.035+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-16T00:23:47.042+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
